assets/lokomotive-kubernetes/bootkube/outputs.tf

```diff
@@ -80,7 +80,7 @@ output "kubernetes_values" {
 }
 
 output "kubelet_values" {
-  value = data.template_file.kubelet.rendered
+  value = data.template_file.kubelet[*].rendered
```

Comment:
Huh, what does this return? :thinking: Random element? Any element, an array? :smile: I see this notation for the first time.

Reply:
How can I check this?

Reply:
E.g. with putting it in `output` of root module (top level).

Reply:
```Terraform will perform the following actions:

  # null_resource.foo[0] will be created
  + resource "null_resource" "foo" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

null_resource.foo[0]: Creating...
null_resource.foo[0]: Creation complete after 0s [id=8484782602377834647]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

foo = [
  "8484782602377834647",
]
``` it gives a slice it seems. and we should be returning string here.

Reply:
But then how does it work? If output is garbled?

Reply:
This is what I found from the tfstate file:


```json
    "kubelet_values": {
      "value": [
        "image: k8s.gcr.io/hyperkube:v1.18.2\nclusterDNS: 10.3.0.10\nclusterDomain: cluster.local\n"
      ],
```

Reply:
By default we don't do upgrades of self-hosted kubelet, as they tend to work unreliably (see #110). If you run with `--upgrade-kubelets`, it will most likely fail, as output will be in the wrong format.

Reply:
When I hardcode the value there like this `value = data.template_file.kubelet[0].rendered`, it fails when the kubelet is disabled as follows:

```bash
module.packet-suraj-lk-cluster.data.template_file.etcds[0]: Refreshing state...

Error: Invalid index

  on ../lokomotive-kubernetes/bootkube/outputs.tf line 83, in output "kubelet_values":
  83:   value = data.template_file.kubelet[0].rendered
    |----------------
    | data.template_file.kubelet is empty tuple

The given key does not identify an element in this collection value.

FATA[0021] error applying cluster: failed checking execution status: exit status 1  args="[]" command="lokoctl cluster apply"
```

What can I do to make it work?

Reply:
`join("", data.template_file.kubelet.*.rendered)` should work.

Reply:
I think this should work `value = join("", data.template_file.kubelet.*.rendered)`.

Reply:
done